internal/rateapi/rateapi.go

```diff
@@ -9,51 +9,64 @@ import (
 	"time"
 )
 
-const url = "https://api.coinbase.com/v2/prices/usd-uah/buy"
+type CoinbaseProvider struct{}
 
-// GetUsdToUahRate returns the current USD to UAH rate
-func GetUsdToUahRate() (float64, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
-	client := &http.Client{}
-	var err error
-
-	req, err := http.NewRequestWithContext(ctx, "GET", url, http.NoBody)
-	if err != nil {
-		return -1, err
-	}
+func NewCoinbaseProvider() *CoinbaseProvider {
+	return &CoinbaseProvider{}
+}
 
-	resp, err := client.Do(req)
-	if err != nil {
-		return -1, err
-	}
-	defer resp.Body.Close()
+const coinbaseURL = "https://api.coinbase.com/v2/prices/usd-uah/buy"
 
-	respBody, err := io.ReadAll(resp.Body)
+// GetUsdToUahRate returns the current USD to UAH rate
+func (cb *CoinbaseProvider) GetUsdToUahRate() (float64, error) {
```

Comment:
Consider generalizing this function and using parameters for base currency and target currency instead